service/s3/api_op_GetBucketLocation.go

```diff
@@ -133,6 +143,62 @@ func addOperationGetBucketLocationMiddlewares(stack *middleware.Stack, options O
 	return nil
 }
 
+type awsRestxml_deserializeOpGetBucketLocation_custom struct {
+}
+
+func (*awsRestxml_deserializeOpGetBucketLocation_custom) ID() string {
+	return "OperationDeserializer"
+}
+
+func (m *awsRestxml_deserializeOpGetBucketLocation_custom) HandleDeserialize(ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler) (
+	out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+) {
+	out, metadata, err = next.HandleDeserialize(ctx, in)
+	if err != nil {
+		return out, metadata, err
+	}
+
+	response, ok := out.RawResponse.(*smithyhttp.Response)
+	if !ok {
+		return out, metadata, &smithy.DeserializationError{Err: fmt.Errorf("unknown transport type %T", out.RawResponse)}
+	}
+
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
+		return out, metadata, awsRestxml_deserializeOpErrorGetBucketLocation(response, &metadata)
+	}
+	output := &GetBucketLocationOutput{}
+	out.Result = output
+
+	var buff [1024]byte
+	ringBuffer := smithyio.NewRingBuffer(buff[:])
+	body := io.TeeReader(response.Body, ringBuffer)
+	rootDecoder := xml.NewDecoder(body)
+	decoder := smithyxml.WrapNodeDecoder(rootDecoder, xml.StartElement{})
+	err = awsRestxml_deserializeOpDocumentGetBucketLocationOutput(&output, decoder)
+	if err == io.EOF {
+		err = nil
+	}
+	if err != nil {
+		var snapshot bytes.Buffer
+		io.Copy(&snapshot, ringBuffer)
+		return out, metadata, &smithy.DeserializationError{
+			Err:      fmt.Errorf("failed to decode response body, %w", err),
+			Snapshot: snapshot.Bytes(),
+		}
+	}
+
+	return out, metadata, err
+}
+
+// $L is helper to swap in a custom deserializer
+func swapDeserializerHelper(stack *middleware.Stack) error {
```

Comment:
nil `$L` here. probably should be `%s` in codegen.

Reply:
I fixed this before, but didn't regenerate the  service. Will do. Thanks